bulk-ingester/src/main/java/org/alfresco/hxi_connector/bulk_ingester/processor/mapper/AlfrescoPropertyMapper.java

```diff
@@ -0,0 +1,201 @@
+ * along with Alfresco. If not, see <http://www.gnu.org/licenses/>.
+ * #L%
+ */
+
+package org.alfresco.hxi_connector.bulk_ingester.processor.mapper;
+
+import static java.lang.String.format;
+import static java.util.Optional.empty;
+import static java.util.Optional.of;
+
+import static org.alfresco.hxi_connector.bulk_ingester.processor.mapper.AlfrescoNodeMapper.CONTENT_PROPERTY;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.ObjectInput;
+import java.io.ObjectInputStream;
+import java.io.Serializable;
+import java.time.ZonedDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+
+import org.alfresco.elasticsearch.db.connector.model.AlfrescoNode;
+import org.alfresco.elasticsearch.db.connector.model.ContentMetadata;
+import org.alfresco.elasticsearch.db.connector.model.PropertyValue;
+import org.alfresco.elasticsearch.db.connector.model.PropertyValueType;
+
+@Slf4j
+@RequiredArgsConstructor
+public class AlfrescoPropertyMapper
+{
+    private final AlfrescoNode alfrescoNode;
+    private final String propertyName;
+
+    public Optional<Map.Entry<String, Serializable>> performMapping()
+    {
+        /* Properties might be duplicated - for example they can have different locale (we can have two descriptions with "en_US_" and "en_UK_" locale) and in this case we want to process them together */
+        List<Serializable> propertyValues = alfrescoNode.getNodeProperties().stream()
+                .filter(nodeProperty -> nodeProperty.getPropertyKey().getLocalName().equals(propertyName))
+                .map(nodeProperty -> mapPropertyValue(nodeProperty.getPropertyValue()))
+                .flatMap(Optional::stream)
+                .toList();
+
+        if (propertyValues.isEmpty())
+        {
+            return empty();
+        }
+        else if (propertyValues.size() == 1)
+        {
+            return of(Map.entry(propertyName, propertyValues.get(0)));
+        }
+        else
+        {
+            return of(Map.entry(propertyName, (Serializable) propertyValues));
+        }
+    }
+
+    private Optional<Serializable> mapPropertyValue(PropertyValue propertyValue)
+    {
+        try
+        {
+            return getPersistedValue(propertyValue).flatMap(persistedValue -> switch (PropertyValueType.getPropertyValueType(propertyValue.getActualType()))
+            {
+            case CONTENT_DATA_ID -> getContentValue(persistedValue);
+            case DATE -> getDateValue(persistedValue);
+            case NODEREF -> getNodeRefValue(persistedValue);
+            default -> of(persistedValue);
+            });
+        }
+        catch (Exception e)
+        {
+            log.warn(format("Error occurred while trying to map property %s value %s of node %s", propertyName, propertyValue, alfrescoNode.getId()), e);
```

Comment:
```suggestion
            log.warn("Error occurred while trying to map property {} value {} of node {}", propertyName, propertyValue, alfrescoNode.getId(), e);
```

Reply:
hmmm, I just checked it and it actually works - thanks for suggestion!

Reply:
fixed!